src/lib/index/server.js

```diff
@@ -25,236 +25,212 @@
 
-	emitter = new EventEmitter(),
+	expressRouter = express.Router,
 
-	ERROR_EVENT = 'error_event',
-	INFO_EVENT = 'info_event',
+	PARAMETER_API_SCHEMA_ENDPOINT = '/:schemaName',
 
-	api = (path, schemaNameToDefinition, publisher, responseWriter = (err, response, context) => {
-		if (err) {
-			response.status(400).send(err.message);
-		} else {
-			response.status(200).send('Ok.');
-		}
-	}) => (request, response) => {
+	PROPERTY_PUBLISHER = 'publisher',
+	PROPERTY_ROUTE_CONFIGURATION = 'route_configuration',
+	PROPERTY_ROUTER_CONFIGURATION = 'router_configuration',
+	PROPERTY_ROUTE_VALIDATOR = 'route_validator',
+	PROPERTY_SERVER = 'server',
 
-		const
-			schemaName = request.params.schemaName,
-			schema = schemaNameToDefinition[schemaName],
-			body = request.body,
-			orizuru = request.orizuru;
-
-		if (!schema) {
-			catchEmitReject(`No schema for '${path}/${schemaName}' found.`, ERROR_EVENT, emitter)
-				.catch(err => {
-					responseWriter(err, response, orizuru);
-				});
-		} else {
-			catchEmitReject(publisher.publish({
-				schema: schema,
-				message: body,
-				context: orizuru
-			}), ERROR_EVENT, emitter).then(() => {
-				responseWriter(null, response, orizuru);
-			}).catch(err => {
-				responseWriter(err, response, orizuru);
-			});
-		}
-
-	};
+	ERROR_EVENT = 'error_event',
+	INFO_EVENT = 'info_event';
 
 /** 
- * Class representing a server. 
+ * The Server for creating routes in a web dyno based on Avro schemas. 
+ * Messages are consumed by Handler
  * 
- * @property {EventEmitter} emitter
+ * @extends EventEmitter
  * @property {string} emitter.ERROR - the error event name
  * @property {string} emitter.INFO - the info event name
  **/
-class Server {
+class Server extends EventEmitter {
 
 	/**
-	 * Constructs a new 'Server'
+	 * Constructs a new 'Server'.
 	 * 
-	 * @param {object} config - { transport [, transportConfig] }
-	 * @param {transport} config.transport - the transport object
-	 * @param {object} config.transportConfig - config for the transport object
-	 * @returns {Server}
+	 * @example 
+	 * const server = new Server();
+	 * @param {Object} config - The server configuration.
+	 * @param {Transport} config.transport - The transport object.
+	 * @param {Object} config.transportConfig - The config for the transport object.
+	 * @returns {Server} - The server.
 	 */
 	constructor(config) {
 
-		// create publisher
-		catchEmitThrow(() => {
-			publisherStore[this] = new Publisher(config);
-		}, ERROR_EVENT, emitter);
+		super();
 
-		// create server
-		const server = express();
+		const me = this;
+		me.info('Creating server.');
 
-		// body parser
-		server.use(bodyParser.json());
+		try {
 
-		// Header security
-		server.use(helmet());
+			// Validate the config
+			new ServerValidator(config);
 
-		// add server to private store
-		serverStore[this] = server;
+			// Create the express server
+			const server = express();
 
-	}
+			// Configure the server
```

Comment:
We should not be adding any middleware here. These could be added to the routes added but probably better to be completely un opinionated. The problem is that this middleware will go off first on all requests. So DOS attacks sending huge payloads could be crafted and bodyparser will be very busy.

Reply:
It seems that express now has a json middleware built in too.

Reply:
Done